### PR TITLE
Fixes OSError Inappropriate ioctl for device

### LINKED ISSUE #2210: timeit - OSError Inappropriate ioctl for device
A new introduced issue.

```
$ timeit ls
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 12437, in run
    r = self.f(self.args, sp_stdin, sp_stdout, sp_stderr, spec)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 12266, in proxy_two
    return f(args, stdin)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 13794, in timeit_alias
    time_number = timer.timeit(number)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 13747, in timeit
    timing = self.inner(it, self.timer)
  File "<xonsh-timeit>", line 6, in inner
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 18340, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 18281, in run_subproc
    command = HiddenCommandPipeline(specs)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 12859, in __init__
    if update_fg_process_group(pipeline_group, background):
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 12795, in update_fg_process_group
    return give_terminal_to(pipeline_group)
  File "/usr/local/lib/python3.5/dist-packages/xonsh/__amalgam__.py", line 8901, in give_terminal_to
    os.tcsetpgrp(sys.stderr.fileno(), pgid)
OSError: [Errno 25] Inappropriate ioctl for device
```

### PR DESCRIPTION
**Pull request by mitnk**
closes #2210

### CODE REVIEW

File: news/term-fd.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+**Changed:** None
+
+**Deprecated:** None
+
+**Removed:** None
+
+**Fixed:**
+
+* Fixed issues like `timeit ls` causing OSError - "Inappropriate ioctl
```
Comment (scopatz) [inline news/term-fd.rst]: Need double backticks here

**Comment by scopatz** [comment]
Thanks for putting this in @mitnk!

**Comment by mitnk** [comment]
just updated a bit

File: news/term-fd.rst (hunk 1 above)
Comment (mitnk) [inline news/term-fd.rst]: fixed.

**Comment by scopatz** [comment]
Thanks @mitnk!

**Pull request merged**